### PR TITLE
Provide System.CodeDom package readme

### PR DESCRIPTION
**Pull request by eNeRGy164**
Fill out details for the `System.CodeDom` NuGet package readme.

See https://github.com/dotnet/runtime/issues/99358

### CODE REVIEW

**Comment by teo-tsirpanis** [comment]
Should it say that CodeDom is superseded by Roslyn?

**Comment by steveharter** [comment]
> Should it say that CodeDom is superseded by Roslyn?

I think it would good to mentioned that for new work, Roslyn should be used (in general, if you don't need CodeDom-specific functionality). Also, we have the readme.md file -- perhaps that would be a better place or an additional place for that statement.

File: src/libraries/System.CodeDom/src/PACKAGE.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
+## About
+
+<!-- A description of the package and where one can find more documentation -->
+
+Provides functionality for dynamically generating and compiling source code using the Code Document Object Model (CodeDOM).
+
+It allows developers to represent code in a language-agnostic format and then generate code in multiple languages, such as C# and VB.NET.
+The primary use cases include creating dynamic code generation tools, runtime code generation, and facilitating code analysis or transformation.
+
```
Comment (buyaa-n) [inline src/libraries/System.CodeDom/src/PACKAGE.md:9]: Maybe add about Roslyn here? Something like:  
```suggestion

For a new modern development consider using Roslyn SDK, check [Cross-Platform Code Generation with Roslyn and .NET Core](https://learn.microsoft.com/archive/msdn-magazine/2017/may/net-core-cross-platform-code-generation-with-roslyn-and-net-core) and [Get started with source generators](https://learn.microsoft.com/dotnet/csharp/roslyn-sdk/source-generators-overview#get-started-with-source-generators) for more info.

```

**Comment by buyaa-n** [comment]
The changes looks good once the feedback about Roslyn is resolved

File: src/libraries/System.CodeDom/src/PACKAGE.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+For a new modern development consider using the [.NET Compiler Platform SDK](https://learn.microsoft.com/otnet/csharp/roslyn-sdk/), check [Cross-Platform Code Generation with Roslyn and .NET Core](https://learn.microsoft.com/archive/msdn-magazine/2017/may/net-core-cross-platform-code-generation-with-roslyn-and-net-core) and [Get started with source generators](https://learn.microsoft.com/dotnet/csharp/roslyn-sdk/source-generators-overview#get-started-with-source-generators) for more info.
```
Comment (buyaa-n) [inline src/libraries/System.CodeDom/src/PACKAGE.md:10]: NIT: typo, and removed the doc link that probably outdated, 
```suggestion
For a new modern development consider using the [.NET Compiler Platform SDK](https://learn.microsoft.com/dotnet/csharp/roslyn-sdk/), in particular [Roslyn source generators](https://learn.microsoft.com/dotnet/csharp/roslyn-sdk/source-generators-overview#get-started-with-source-generators).
```

**Review by buyaa-n — APPROVED** [review]
LGTM, thanks you!

**Pull request merged**